libsolidity/lsp/Utils.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <liblangutil/SourceLocation.h>
+
+#include <libsolidity/ast/ASTForward.h>
+
+#include <libsolutil/JSON.h>
+
+#include <optional>
+#include <vector>
+
+namespace solidity::langutil
+{
+class CharStreamProvider;
+}
+
+namespace solidity::lsp
+{
+
+class FileRepository;
+
+std::optional<langutil::LineColumn> parseLineColumn(Json::Value const& _lineColumn);
+Json::Value toJson(langutil::LineColumn _pos);
+Json::Value toJsonRange(langutil::LineColumn const& _start, langutil::LineColumn const& _end);
+
+Json::Value toRange(
+	langutil::CharStreamProvider const& _charStreamProvider,
+	langutil::SourceLocation const& _location
+);
+
+Json::Value toJson(
```

Comment:
I think it could be very cumbersome to pass `charStreamProvider` along all the way. Can't this be a nice util function inside the Handler base class?